source/fundamentals/connection/mongoclientsettings.txt

```diff
@@ -0,0 +1,198 @@
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin SocketSettings
+   :end-before: end SocketSettings
+   :language: java
+   :emphasize-lines: 3-5
+   :dedent:
+
+For a list of methods you can use to specify your socket settings, see
+the `API Documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/SocketSettings.Builder.html>`__.
+
+.. _mcs-connectionpool-settings:
+
+Connection Pool Settings
+------------------------
+
+Chain the `applyToConnectionPoolSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToConnectionPoolSettings(com.mongodb.Block)>`__
+method to modify the way the driver manages its' connection pool.
+
+Example
+~~~~~~~
+
+This example specifies the following driver behavior in a pool of
+``Connection`` types:
+
+- The thread to wait at most ``3 MINUTES`` for an available connection
+- To have at most ``200`` connections associated with the pool
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin ConnectionPoolSettings
+   :end-before: end ConnectionPoolSettings
+   :language: java
+   :emphasize-lines: 3-5
+   :dedent:
+
+For a list of methods you can use to specify your connection pool
+settings, see the `API Documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/ConnectionPoolSettings.Builder.html>`__.
+
+.. _mcs-server-settings:
+
+Server Settings
+---------------
+
+Chain the `applyToServerSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToServerSettings(com.mongodb.Block)>`__
+method to modify the driver's behavior when monitoring each MongoDB
+server.
+
+Example
+~~~~~~~
+
+This example specifies the following driver behavior in a MongoDB server:
+
+- To interval between server monitoring checks to be at least ``700 MILLISECONDS`` 
+- The cluster monitor to attempt reaching a server every ``15 SECONDS``
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin ServerSettings
+   :end-before: end ServerSettings
+   :language: java
+   :emphasize-lines: 3-5
+   :dedent:
+
+For a list of methods you can use to specify your server settings, see
+the `API Documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/ServerSettings.Builder.html>`__.
+
+.. _mcs-ssl-settings:
+
+TLS/SSL Settings
+----------------
+
+Chain the `applyToSslSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToSslSettings(com.mongodb.Block)>`__
+method to modify the driver's behavior when using TLS/SSL to secure a
+connection between your application and MongoDB.
+
+Example
+~~~~~~~
+
+This example specifies the following driver behavior for connecting to
+MongoDB with TLS/SSL:
+
+- To ensure your Atlas connection string works
+- To not allow a mismatched ``hostname`` in the server and certificate
```

Comment:
This might mislead users into thinking that the default behavior is to allow this mismatch, whereas the default is actually to now allow it.  Especially since this is a security-sensitive area, we should be very clear.  On the other hand, I don't want the example to be one that weakens security.

Perhaps use SSLContext configuration as the example?